crates/aptos/src/common/types.rs

```diff
@@ -1022,6 +1022,41 @@ impl RestOptions {
     }
 }
 
+/// Options for optimization level
+#[derive(Debug, Clone, Parser)]
+pub enum OptimizationLevel {
+    /// No optimizations
+    None,
+    /// Default optimization level
+    Standard,
+    /// Extra optimizations (-O3)
+    Full,
+}
+
+impl Default for OptimizationLevel {
+    fn default() -> Self {
+        Self::Standard
+    }
+}
+
+impl FromStr for OptimizationLevel {
+    type Err = anyhow::Error;
+
+    /// Parses a language version. If the caller only provides a major
+    /// version number, this chooses the latest stable minor version (if any).
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_uppercase().as_str() {
+            "NONE" => Ok(Self::None),
+            "STANDARD" => Ok(Self::Standard),
+            "FULL" => Ok(Self::Full),
+            _ => bail!(
+                "unrecognized optimization level `{}` (supported versions: `none`, `standard`, `full`)",
+                s
+            ),
+        }
+    }
```

Comment:
Everything else I believe went by lowercase instead of uppercase but I may be wrong

Reply:
Fine.